src/crimson/os/seastore/seastore_types.h

```diff
@@ -1347,8 +1347,6 @@ std::ostream &operator<<(std::ostream &out, mod_time_point_printer_t tp);
 constexpr sea_time_point
 get_average_time(const sea_time_point& t1, std::size_t n1,
                  const sea_time_point& t2, std::size_t n2) {
-  assert(t1 != NULL_TIME);
-  assert(t2 != NULL_TIME);
```

Comment:
These are still valid and shouldn't be removed, right?

Reply:
I called `get_average_time()` and `segments_info_t::update_modify_time()` no matter whether `num_extents` is zero or not, which led to violation of these asserts.

Reply:
Maybe `get_averge_time()` and `segments_info_t::update_modify_time()` only needs to be called in cases when `num_extents` is non-zero, am I right?

Reply:
The original asserts were working, and looks `segments_info_t::update_modify_time()` was already correct and reasonable.

For async_cleaner.cc 1388, How about:
```c++
if (num_extents == 0) {
  ceph_assert(tail_time == NULL_TIME);
  // pass
} else {
  ceph_assert(header_time != NULL_TIME);
  ceph_assert(tail_time != NULL_TIME);
  sea_time_point avg_time = get_average_time(
	    header_time, 1, tail_time, 1);
  segments.update_modify_time(segment_id, avg_time, num_extents);
}
```

Reply:
Anyway, it's meaningless to get average from a NULL_TIME.

Reply:
Done:-)

Reply:
@xxhdx1985126 forget to remove them? Others LGTM.

Reply:
Ah, yes, sorry.